torch/_export/__init__.py

```diff
@@ -559,6 +559,84 @@ def export(
         preserve_module_call_signature=preserve_module_call_signature,
     )
 
+
+def _prepare_module(
+    gm_torch_level: torch.fx.GraphModule,
+    aot_export_args
+) -> List[str]:
+    flat_args = pytree.tree_leaves(aot_export_args)
+    user_input_names = []
+    with gm_torch_level.graph.inserting_before():
+        for i, (arg, node) in enumerate(zip(flat_args, gm_torch_level.graph.nodes)):
+            assert node.op == "placeholder"
+            user_input_names.append(node.name)
+            if isinstance(arg, torch.Tensor):
+                assert not hasattr(gm_torch_level, node.name)
+                gm_torch_level.register_buffer(node.name, arg)
+                node.replace_all_uses_with(gm_torch_level.graph.get_attr(node.name))
+    for node in list(gm_torch_level.graph.nodes):
+        if node.op == "placeholder":
```

Comment:
Why would there be extra placeholders?

Reply:
because replace_all_uses_with() doesn't remove the replaced nodes.

Reply:
I think you can just call graph.erase_node(node_you_want_to_delete) right after replace_all_uses_with ?